.gitignore

```diff
@@ -22,3 +22,7 @@ develop-eggs
 .installed.cfg
 lib
 lib64
+.asv
+.benchmarks
+.ipynb_checkpoints
+tags*
```

Comment:
What generates the `tags*`?

Reply:
`tags` is employed by [ctags](https://github.com/universal-ctags/ctags). I decided to include it here as `.vscode/` and  `/DS_Store` where also in this file although it is something probably employed only by me. Should I remove it?

Reply:
If we're not using ctags along with our code anywhere, I would leave it out.

Reply:
`tags*` is a pattern which seems like it could easily have conflicts with stuff which could be useful.  For example, it would match `tags.py` too.  At the absolute _most_ it should be `tags`/`TAGS` for ctags, but a better solution would probably be to configure your `ctags` to use a better filename, or add the file to your personal ignore files.  You can configure your local git with the variable `core.excludesFile` and point it to somewhere that has you specific files in it, or you can do it on a repository-by-repository basis by adding lines to the file `.git/info/exlude`.

Generally, the only things which should be in a repository's `.gitignore` should be files to be ignored that are actually generated as part of the build process.  However, it can often be nice to add hidden files that are really common to many editors, and have sensible, nonconflicting names, because it just eases the burden for contributors and maintainers.